feat(browser): Add offline transport docs (#6292)
getsentry/sentry-docs: src/platforms/javascript/common/configuration/transports.mdx

---
title: Transports
sidebar_order: 30
description: "Transports let you change the way in which events are delivered to Sentry."
notSupported:
  - javascript.capacitor
  - javascript.cordova
  - javascript.electron
  - javascript.wasm
---

> The JavaScript SDK uses a `transport` to send events to Sentry. On modern
> browsers, the default transport uses the browsers `fetch` API to send events.
> 
> The default transports drop events if they fail to send due to a lack of connection.

Reviewer: ```suggestion

The JavaScript SDK uses a `transport` to send events to Sentry. On modern browsers, most transports use the browsers' `fetch` API to send events. Transports will drop an event if it fails to send due to a lack of connection.
```
Let me know if this changes the meaning too much. If it's important to differentiate between a transport and a "default" transport, it might be good to explain the difference a bit more :)

### Offline Caching

> If your JavaScript application is designed to continue working offline,
> dropping events when no connection is available could mean you are missing
> events that occur in that mode.
> 
> To enable offline caching of events, `makeBrowserOfflineTransport` can be used
> to wrap existing transports and queue events using the browsers IndexedDB storage.
> Sending of events is then retried later.
> 
> <PlatformContent includePath="configuration/offline" />
> 
> This adds a number of optional properties to `transportOptions`:

Reviewer: ```suggestion
If your JavaScript application is designed to continue working offline,
dropping events when no connection is available and missing
offline events, could mean you're missing important information.

To enable offline events caching, use `makeBrowserOfflineTransport` to wrap existing transports and queue events 
using the browsers' IndexedDB storage. Once your application comes back online, all events will be sent together.

<PlatformContent includePath="configuration/offline" />

Here are a few optional properties you can add to `transportOptions`:
```

Again, please let me know if I've changed the meaning too much.
